src/Stratis.Bitcoin/Configuration/NodeSettings.cs

```diff
@@ -132,14 +135,28 @@ public static NodeSettings Default(Network network = null, ProtocolVersion proto
         }
 
         /// <summary>
-        /// Initializes configuration from command line arguments.
-        /// <para>This includes loading configuration from file.</para>
+        /// Processes the command line arguments only.
         /// </summary>
         /// <param name="args">Application command line arguments.</param>
-        /// <returns>Initialized node configuration.</returns>
-        /// <exception cref="ConfigurationException">Thrown in case of any problems with the configuration file or command line arguments.</exception>
-        public NodeSettings LoadArguments(string[] args)
+        /// <param name="delayedProcessing">Set to <c>true</c> if the configuration file should not be read yet.</param>
+        /// <returns>Node configuration with arguments recorded.</returns>
+        /// <exception cref="ConfigurationException">Thrown in case of any problems with the command line arguments.</exception>
+        public NodeSettings LoadArgumentsDelayed(string[] args)
```

Comment:
OK good stuff, we are getting closer to a good solution here!

So I just realized that seeing as both LoadArguments And LoadConfiguration return "this" we can build this as a fluent interface now. i.e.

If you wanna load just the arguments, for example:

`this.nodeSettings.LoadArguments(args);`

If you wanna load the arguments as well as the Configuration, you can call it like this:

`this.nodeSettings.LoadArguments(args).LoadConfiguration();`

I think this will be even cleaner!

I.e. LoadArguments ONLY loads arguments and LoadConfiguration ONLY loads config but in a fluent interface you chain these methods together.

Reply:
Ok done.